apps/server/src/modules/system/repo/mikro-orm/system.repo.spec.ts

```diff
@@ -171,6 +170,43 @@ describe(SystemRepo.name, () => {
 		});
 	});
 
+	describe('getSystemsByIds', () => {
+		describe('when no system exists', () => {
+			it('should return empty array', async () => {
+				const result = await repo.getSystemsByIds([new ObjectId().toHexString()]);
+
+				expect(result).toEqual([]);
+			});
+		});
+
+		describe('when different systems exist', () => {
+			const setup = async () => {
+				const system1: SystemEntity = systemEntityFactory.buildWithId();
+				const system2: SystemEntity = systemEntityFactory.buildWithId();
+				const system3: SystemEntity = systemEntityFactory.buildWithId();
+
+				await em.persistAndFlush([system1, system2, system3]);
+				em.clear();
+
+				const system1Props = SystemEntityMapper.mapToDo(system1);
+				const system2Props = SystemEntityMapper.mapToDo(system2);
+				const system3Props = SystemEntityMapper.mapToDo(system3);
+
+				const expectedSystems = [system1Props, system2Props, system3Props];
+
+				return { expectedSystems };
+			};
+
+			it('should return the systems', async () => {
+				const { expectedSystems } = await setup();
+
+				const result = await repo.getSystemsByIds(expectedSystems.map((s) => s.id));
```

Comment:
expectedSystems.map((s) => s.id) -> in setup und expectedSystemIds hier verwenden, oder zumindest in die Zeile drüber auf eine const.